File: java/169.java
class Solution {
    public int majorityElement(int[] nums) {
        // sorting is faster in real practise
        // int l = nums.length;
        // Arrays.sort(nums);
        // return nums[l/2];
        int count = 0;
        Integer cand = null;
        for(int num:nums){
            if(count == 0){
                cand = num;
            }
            count += (cand == num) ? 1 : -1;
        }
        return cand;
    }
}

File: go/169.go
func majorityElement(nums []int) int {
	// sort.Ints(nums)
	// return nums[len(nums)/2]
	count := 0
	var cand int
	for _, num := range nums {
		if count == 0 {
			cand = num
		}
		if cand == num {
			count++
		} else {
			count--
		}
	}
	return cand
}